### PR TITLE
Position iterator on Next() if neccessary

### LINKED ISSUE #702: `db.Iterator` is always invalid if `Seek` is never called
`NewIterator` returns an invalid Iterator.

```
// NewIter returns an iterator that is unpositioned (Iterator.Valid() will
// return false). The iterator can be positioned via a call to SeekGE,
// SeekPrefixGE, SeekLT, First or Last. Only indexed batches support iterators.
```

### PR DESCRIPTION
**Pull request by omerfirmak**
Fixes #702

### CODE REVIEW

**Review by jelilat — APPROVED** [review]

**Comment by IronGauntlets** [comment]
I don't think I understand the problem that is being solved here. Are we saying that we want to be able to `Next()` on a new iterator without calling seek?

**Comment by omerfirmak** [comment]
> I don't think I understand the problem that is being solved here. Are we saying that we want to be able to call `Next()` on a new iterator without calling `Seek()`?

Yes

File: db/pebble/db_test.go [hunk 1]
```diff
@@ -350,3 +350,44 @@ func TestPrefixSearch(t *testing.T) {
...
+		})
+
+		t.Run("Next() should validate iterator", func(t *testing.T) {
+			assert.True(t, it.Next())
+		})
+
+		require.NoError(t, it.Close())
+	})
+
+	t.Run("first Next() should work as expected after a Seek()", func(t *testing.T) {
```
Comment (IronGauntlets) [inline db/pebble/db_test.go:383]:
```suggestion
	t.Run("Next() should work as expected after a Seek()", func(t *testing.T) {
```

File: db/pebble/db_test.go [hunk 2]
```diff
@@ -350,3 +350,44 @@ func TestPrefixSearch(t *testing.T) {
...
+		it, err := txn.NewIterator()
+		require.NoError(t, err)
+
+		require.True(t, it.Seek([]byte{0}))
+		require.True(t, it.Next())
+		require.Equal(t, []byte{1}, it.Key())
```
Comment (IronGauntlets) [inline db/pebble/db_test.go:389]: small nitpick: should we also check the value?

Comment (omerfirmak) [reply db/pebble/db_test.go:389]: not really, that would be testing the underlying db implementation. Not the change here.

File: db/pebble/db_test.go (hunk 1 above)
Comment (omerfirmak) [reply db/pebble/db_test.go]: done

**Review by IronGauntlets — APPROVED** [review]

**Pull request merged**